pkg/api/management/api/logging/list_handler.go

```diff
@@ -0,0 +1,112 @@
+package logging
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/convert"
+	loggingconfig "github.com/rancher/rancher/pkg/logging/config"
+	corev1 "github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ClusterLoggingHandler struct {
+	ClusterLoggingClient v3.ClusterLoggingInterface
+	CoreV1               corev1.Interface
+}
+
+func (h *ClusterLoggingHandler) ListHandler(request *types.APIContext, _ types.RequestHandler) error {
+	if request.ID != "" {
+		strArray := strings.Split(request.ID, ":")
+		if len(strArray) == 2 {
+			clusterName := strArray[0]
+			id := strArray[1]
+			cl, err := h.ClusterLoggingClient.GetNamespaced(clusterName, id, metav1.GetOptions{})
+			if err != nil {
+				logrus.Errorf("get cluster logging by ID failed, %v", err)
+				return httperror.NewAPIError(httperror.ServerError, fmt.Sprintf("%v", err))
+			}
+			if cl.Spec.EmbeddedConfig != nil {
+				err = h.setEmbeddedEndpoint(cl)
+				if err != nil {
+					return err
```

Comment:
any reason why sometimes we return httperror, and sometimes just error?

Reply:
In the API layer if you return err (not httperror) the client will just get a 500